engines/native/resultset.go

```diff
@@ -140,6 +140,12 @@ func (r *resultSet) ExtractFolder(path string, handler engines.FileHandler) erro
 }
 
 func (r *resultSet) Dispose() error {
+	// if we didn't create a user for this sandbox, we shouldn't destroy it either.
+	// TODO: why is this cleanup here, and not in Sandbox?
```

Comment:
Because we need to user folder to persist after task is done to extract artifacts after `WaitForResult` returns.